### PR TITLE
CLI tools: discover commands in plugins that are enabled indirectly as dependencies; rework plugin state assertions

### PR DESCRIPTION
**Pull request by michaelklishin**
This is #6289 by @SimonUnge (AWS) with some test suite improvements from me.

### CODE REVIEW

**Comment by michaelklishin** [comment]
I can reproduce the CI failure locally with both Bazel and Make:

``` shell
bazel test deps/rabbit/feature_flags_with_unpriveleged_user_SUITE
```

Investigating.

**Comment by michaelklishin** [comment]
From @pjk25:

So, I believe the issue is this. The test helpers that start nodes always invoke `rabbitmqctl` with no args as part of the setup assertions. We have always expected this to just print the help.
The PR adds a call to `PluginsHelpers.list(opts)` in `deps/rabbitmq_cli/lib/rabbitmq/cli/core/command_modules.ex` at line 77. This ends checking if the plugins dir exists synchronously. Since bazel always has an empty temp dir for the node when things start, the plugins dir does not yet exist. Since Make reuses directories, we probably don’t hit it.
Probably the lack of the directory should just be treated as empty, with no extra plugins, at this point in time when re run rabbitmqctl. The other option would be to adjust either the `rabbitmq_ct_*helpers` to create the directory, or `rabbitmq-run.sh`, if we know that all the rabbitmq installers always create the directory.

File: deps/rabbitmq_cli/lib/rabbitmq/cli/core/command_modules.ex [hunk 1]
```diff
@@ -74,8 +74,31 @@ defmodule RabbitMQ.CLI.Core.CommandModules do
           []
       end
 
+    all_plugins = PluginsHelpers.list(opts)
```
Comment (michaelklishin) [inline deps/rabbitmq_cli/lib/rabbitmq/cli/core/command_modules.ex:77]: This change means we either have to always have the plugins directory set, including in all kinds of test environments, or avoid returning an error when the directory does not exist.

Comment (michaelklishin) [reply deps/rabbitmq_cli/lib/rabbitmq/cli/core/command_modules.ex:77]: c2f87ad2d7cec6db2718cc8bcd99177a885ff85d and 0c821ded1a1b4a45d1f98387a4abd15db2757e39 are two different solutions. We are discussing with @pjk25 if we should apply both or just one of them.

**Comment by pjk25** [comment]
@michaelklishin let's revert https://github.com/rabbitmq/rabbitmq-server/pull/6316/commits/c2f87ad2d7cec6db2718cc8bcd99177a885ff85d and see it that passes. It seems to interfere with a few of the feature_flags suites which expect to load some additional test plugins from other paths.
If it passes, we can consider if we want to be more judicious injecting the env var, or take a look at the predence of vars in the failing tests.

**Pull request merged**